File: sdk-python/examples/content_quality.py
"""Demonstrates content quality scanning with C6 Guardrails.

Shows:
- Evaluating a prompt and response pair for groundedness, relevance,
  and hallucination scores.
- Printing the verdict and individual score arrays.
- Using asyncio.run().
"""

import asyncio

from c6_guardrails import GatewayClient, GatewayClientOptions


async def main() -> None:
    client = GatewayClient(GatewayClientOptions(base_url="http://localhost:8082", api_key="ak_..."))

    result = await client.cq_scan(
        input="What is the capital of France?",
        response="The capital of France is Paris.",
    )

    print(f"Verdict: {result.verdict}")
    print(f"Groundedness: {result.groundedness}")
    print(f"Relevance: {result.relevance}")
    print(f"Hallucination: {result.hallucination}")
    print(f"Action: {result.action}")
    print(f"Reason: {result.reason}")

    await client.close()


asyncio.run(main())

File: sdk-node/examples/content-quality.ts
/**
 * Content quality scan example — demonstrates cqScan() with a
 * prompt/response pair, printing groundedness, relevance, and
 * hallucination scores, and handling the verdict field.
 */

import { GatewayClient } from '@c6web/guardrails';

const client = new GatewayClient();

const result = await client.cqScan(
  'What is the capital of France?',
  'The capital of France is Paris.',
);

console.log('Verdict:', result.verdict);
console.log('Action:', result.action);
console.log('Groundedness:', result.groundedness);
console.log('Relevance:', result.relevance);
console.log('Hallucination:', result.hallucination);
console.log('Reason:', result.reason);
